pkg/test/client_test.go

```diff
@@ -81,41 +128,57 @@ func TestNewClient(t *testing.T) {
 
 	t.Run("mock List", func(t *testing.T) {
 		defer func() { fclient.MockList = nil }()
-		fclient.MockList = func(ctx context.Context, opts *client.ListOptions, list runtime.Object) error {
+		fclient.MockList = func(ctx context.Context, list runtime.Object, opts ...client.ListOption) error {
 			return expectedErr
 		}
-		assert.EqualError(t, fclient.List(context.TODO(), &client.ListOptions{Namespace: "somenamespace"}, &v1.SecretList{}), expectedErr.Error())
+		assert.EqualError(t, fclient.List(context.TODO(), &v1.SecretList{}, client.InNamespace("somenamespace")), expectedErr.Error())
 	})
 
 	t.Run("mock Create", func(t *testing.T) {
 		defer func() { fclient.MockCreate = nil }()
-		fclient.MockCreate = func(ctx context.Context, obj runtime.Object) error {
+		fclient.MockCreate = func(ctx context.Context, obj runtime.Object, option ...client.CreateOption) error {
 			return expectedErr
 		}
 		assert.EqualError(t, fclient.Create(context.TODO(), &v1.Secret{}), expectedErr.Error())
 	})
 
 	t.Run("mock Update", func(t *testing.T) {
 		defer func() { fclient.MockUpdate = nil }()
-		fclient.MockUpdate = func(ctx context.Context, obj runtime.Object) error {
+		fclient.MockUpdate = func(ctx context.Context, obj runtime.Object, option ...client.UpdateOption) error {
 			return expectedErr
 		}
 		assert.EqualError(t, fclient.Update(context.TODO(), &v1.Secret{}), expectedErr.Error())
 	})
 
 	t.Run("mock Delete", func(t *testing.T) {
 		defer func() { fclient.MockDelete = nil }()
-		fclient.MockDelete = func(ctx context.Context, obj runtime.Object, opts ...client.DeleteOptionFunc) error {
+		fclient.MockDelete = func(ctx context.Context, obj runtime.Object, opts ...client.DeleteOption) error {
 			return expectedErr
 		}
 		assert.EqualError(t, fclient.Delete(context.TODO(), &v1.Secret{}), expectedErr.Error())
 	})
 
+	t.Run("mock DeleteAllOf", func(t *testing.T) {
+		defer func() { fclient.MockDeleteAllOf = nil }()
+		fclient.MockDeleteAllOf = func(ctx context.Context, obj runtime.Object, opts ...client.DeleteAllOfOption) error {
+			return expectedErr
+		}
+		assert.EqualError(t, fclient.DeleteAllOf(context.TODO(), &v1.Secret{}), expectedErr.Error())
+	})
+
 	t.Run("mock Status Update", func(t *testing.T) {
 		defer func() { fclient.MockStatusUpdate = nil }()
-		fclient.MockStatusUpdate = func(ctx context.Context, obj runtime.Object) error {
+		fclient.MockStatusUpdate = func(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
 			return expectedErr
 		}
 		assert.EqualError(t, fclient.MockStatusUpdate(context.TODO(), &v1.Secret{}), expectedErr.Error())
 	})
+
+	t.Run("mock Status Patch", func(t *testing.T) {
```

Comment:
There is missing the same subtest for the `MockPatch` method

Reply:
Good catch! Fixed in https://github.com/codeready-toolchain/toolchain-common/pull/48/commits/306f81888e72953684d26874f19edca862f54f94